### DISCUSSION TITLE
MyST vs RMarkdown vs Quarto formats

### DISCUSSION THREAD

**Discussion by echarles** [executablebooks/meta#945] [General]
I am trying to understand the reasoning behind MyST specification (https://github.com/executablebooks/myst-spec), espcially looking at similar initiatives like RMarkdown as the format used by Quarto.

I have not found a clear specification of RMarkdown format (apart from this bookdown page https://bookdown.org/yihui/rmarkdown/markdown-document.html and this RStudio page https://rmarkdown.rstudio.com/authoring_quick_tour.html). The Quarto spec is also sparse (https://quarto.org/docs/authoring/markdown-basics.html). As a user, I have been using bookdown to create PDF with citations...

Is the goal of MyST to bring more clear specifications? How MyST is compatible with other formats? Does MyST add missing features to existing formats?

Users and tool developers may have similar questions. There is a TODO in the myst-spec README repository https://github.com/executablebooks/myst-spec/blob/main/README.md so it may be good to fill that TODO. I post this question in this more general `meta` repository as it is a quite generic question. I can see a risk to bring fragmentation and confusion in the future if the reasoning is not communicated.

```
Relation to other markup languages, frameworks
[TODO]

CommonMark
GitHub Flavored Markdown
Pandoc
Unified
Markdown-It
```

**Comment by chrisjsewell** [comment]
Hey Eric thanks for the feedback, I hope #843 and then things like https://github.com/executablebooks/myst-eps/discussions/4#discussioncomment-3820108 will start to address this.

Very simply IMO:

MyST is targeted at scientific/technical writing, and outputting to multiple formats, with hopefully a community driven/maintained specification

- CommonMark (and GFM): have a clear spec ✅, but limited syntax and only HTML as output format in spec
- Pandoc: has multiple output formats ✅, a larger syntax, but no formal spec

markdown-it and unified are more implementation parsers of a spec (commonmark, gfm, etc)

> Is the goal of MyST to bring more clear specifications? 

yes

> How MyST is compatible with other formats? 

where possible yes, e.g. using commonmark as a "base" spec and then trying to utilise existing "pseudo-standard" extensions to that
although I guess there is no "promise" of cross-compatibility

> Does MyST add missing features to existing formats?

yes; things like captioned figures, cross-document/project referencing, etc

**Comment by echarles** [comment]
Thx Chris. I am putting myself in the shoes of a user who needs to choose a tool. The supported format to author and publish is one the criteria I guess that will be considered, as the interoperability of those. Write research papers in a format that can be used by as much tools should be considered, hence my question.
